frontend/packages/dev-console/src/components/topology/components/GraphComponent.tsx

```diff
@@ -1,21 +1,41 @@
 import * as React from 'react';
+import { connect } from 'react-redux';
+import { RootState } from '@console/internal/redux';
 import { GraphComponent as BaseGraphComponent } from '@console/topology';
+import { TopologyFilters, getTopologyFilters } from '../filters/filter-utils';
 
 type GraphComponentProps = React.ComponentProps<typeof BaseGraphComponent> & {
   dragEditInProgress?: boolean;
+  filters: TopologyFilters;
 };
 
 const DRAG_ACTIVE_CLASS = 'odc-m-drag-active';
+const FILTER_ACTIVE_CLASS = 'odc-m-filter-active';
 
 const GraphComponent: React.FC<GraphComponentProps> = (props) => {
+  const { dragEditInProgress, filters } = props;
   React.useEffect(() => {
-    if (props.dragEditInProgress) {
-      document.body.classList.add(DRAG_ACTIVE_CLASS);
-    } else {
-      document.body.classList.remove(DRAG_ACTIVE_CLASS);
+    const addClassList = [];
+    const removeClassList = [];
+    filters.searchQuery !== ''
+      ? addClassList.push(FILTER_ACTIVE_CLASS)
+      : removeClassList.push(FILTER_ACTIVE_CLASS);
+    dragEditInProgress
+      ? addClassList.push(DRAG_ACTIVE_CLASS)
+      : removeClassList.push(DRAG_ACTIVE_CLASS);
+
+    if (addClassList.length) {
+      addClassList.forEach((className) => document.body.classList.add(className));
+    }
+    if (removeClassList.length) {
+      removeClassList.forEach((className) => document.body.classList.remove(className));
     }
-  }, [props.dragEditInProgress]);
+  }, [dragEditInProgress, filters]);
```

Comment:
Can we use filters.searchQuery? So, It doesn't trigger every time one of the other filters is changed. 
```suggestion
  }, [dragEditInProgress, filters.searchQuery]);
```